In [2]:
import pandas as pd
import math 
import numpy as np
import matplotlib.pyplot as plt

In [26]:
vgs_raw = pd.read_csv('vgsales.csv')

In [27]:
vgs_raw.shape

(16598, 11)

In [28]:
vgs_raw.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 16598 entries, 0 to 16597
Data columns (total 11 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   Rank          16598 non-null  int64  
 1   Name          16598 non-null  object 
 2   Platform      16598 non-null  object 
 3   Year          16327 non-null  float64
 4   Genre         16598 non-null  object 
 5   Publisher     16540 non-null  object 
 6   NA_Sales      16598 non-null  float64
 7   EU_Sales      16598 non-null  float64
 8   JP_Sales      16598 non-null  float64
 9   Other_Sales   16598 non-null  float64
 10  Global_Sales  16598 non-null  float64
dtypes: float64(6), int64(1), object(4)
memory usage: 1.4+ MB


## Null values
### We have two columns that have null values
#### Year & Publisher

In [29]:
## printing the total number of null values
vgs_raw.isnull().sum()

Rank              0
Name              0
Platform          0
Year            271
Genre             0
Publisher        58
NA_Sales          0
EU_Sales          0
JP_Sales          0
Other_Sales       0
Global_Sales      0
dtype: int64

In [49]:
vgs_raw['Year'].isnull()
vgs_raw[vgs_raw['Year'].isnull()]


Rank                         Name Platform  Year         Genre  \
179      180              Madden NFL 2004      PS2   NaN        Sports   
377      378             FIFA Soccer 2004      PS2   NaN        Sports   
431      432   LEGO Batman: The Videogame      Wii   NaN        Action   
470      471   wwe Smackdown vs. Raw 2006      PS2   NaN      Fighting   
607      608               Space Invaders     2600   NaN       Shooter   
...      ...                          ...      ...   ...           ...   
16307  16310                Freaky Flyers       GC   NaN        Racing   
16327  16330                    Inversion       PC   NaN       Shooter   
16366  16369  Hakuouki: Shinsengumi Kitan      PS3   NaN     Adventure   
16427  16430                 Virtua Quest       GC   NaN  Role-Playing   
16493  16496                   The Smurfs      3DS   NaN        Action   

                                    Publisher  NA_Sales  EU_Sales  JP_Sales  \
179                           Electronic Arts      4.26      0.26      0.01   
377                           Electronic Arts      0.59      2.36      0.04   
431    Warner Bros. Interactive Entertainment      1.86      1.02      0.00   
470                                       NaN      1.57      1.02      0.00   
607                                     Atari      2.36      0.14      0.00   
...                                       ...       ...       ...       ...   
16307                                 Unknown      0.01      0.00      0.00   
16327                      Namco Bandai Games      0.01      0.00      0.00   
16366                                 Unknown      0.01      0.00      0.00   
16427                                 Unknown      0.01      0.00      0.00   
16493                                 Unknown      0.00      0.01      0.00   

       Other_Sales  Global_Sales  
179           0.71          5.23  
377           0.51          3.49  
431           0.29          3.17  
470           0.41          3.00  
607           0.03          2.53  
...            ...           ...  
16307         0.00          0.01  
16327         0.00          0.01  
16366         0.00          0.01  
16427         0.00          0.01  
16493         0.00          0.01  

[271 rows x 11 columns]

In [8]:
print(vgs_raw[vgs_raw['Year'].isnull()].Name)

179                  Madden NFL 2004
377                 FIFA Soccer 2004
431       LEGO Batman: The Videogame
470       wwe Smackdown vs. Raw 2006
607                   Space Invaders
                    ...             
16307                  Freaky Flyers
16327                      Inversion
16366    Hakuouki: Shinsengumi Kitan
16427                   Virtua Quest
16493                     The Smurfs
Name: Name, Length: 271, dtype: object


In [40]:
vgs_clean = vgs_raw.copy()

In [33]:
vgs_clean

Rank                                              Name Platform  \
0          1                                        Wii Sports      Wii   
1          2                                 Super Mario Bros.      NES   
2          3                                    Mario Kart Wii      Wii   
3          4                                 Wii Sports Resort      Wii   
4          5                          Pokemon Red/Pokemon Blue       GB   
...      ...                                               ...      ...   
16593  16596                Woody Woodpecker in Crazy Castle 5      GBA   
16594  16597                     Men in Black II: Alien Escape       GC   
16595  16598  SCORE International Baja 1000: The Official Game      PS2   
16596  16599                                        Know How 2       DS   
16597  16600                                  Spirits & Spells      GBA   

         Year         Genre   Publisher  NA_Sales  EU_Sales  JP_Sales  \
0      2006.0        Sports    Nintendo     41.49     29.02      3.77   
1      1985.0      Platform    Nintendo     29.08      3.58      6.81   
2      2008.0        Racing    Nintendo     15.85     12.88      3.79   
3      2009.0        Sports    Nintendo     15.75     11.01      3.28   
4      1996.0  Role-Playing    Nintendo     11.27      8.89     10.22   
...       ...           ...         ...       ...       ...       ...   
16593  2002.0      Platform       Kemco      0.01      0.00      0.00   
16594  2003.0       Shooter  Infogrames      0.01      0.00      0.00   
16595  2008.0        Racing  Activision      0.00      0.00      0.00   
16596  2010.0        Puzzle    7G//AMES      0.00      0.01      0.00   
16597  2003.0      Platform     Wanadoo      0.01      0.00      0.00   

       Other_Sales  Global_Sales  
0             8.46         82.74  
1             0.77         40.24  
2             3.31         35.82  
3             2.96         33.00  
4             1.00         31.37  
...            ...           ...  
16593         0.00          0.01  
16594         0.00          0.01  
16595         0.00          0.01  
16596         0.00          0.01  
16597         0.00          0.01  

[16598 rows x 11 columns]

In [44]:
##vgs_clean.dropna(inplace= True)

vgs_clean = vgs_clean[vgs_clean['Year'].notna()]


In [45]:
vgs_clean.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 16327 entries, 0 to 16597
Data columns (total 11 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   Rank          16327 non-null  int64  
 1   Name          16327 non-null  object 
 2   Platform      16327 non-null  object 
 3   Year          16327 non-null  float64
 4   Genre         16327 non-null  object 
 5   Publisher     16291 non-null  object 
 6   NA_Sales      16327 non-null  float64
 7   EU_Sales      16327 non-null  float64
 8   JP_Sales      16327 non-null  float64
 9   Other_Sales   16327 non-null  float64
 10  Global_Sales  16327 non-null  float64
dtypes: float64(6), int64(1), object(4)
memory usage: 1.5+ MB


In [46]:
vgs_clean.isnull().sum()

Rank             0
Name             0
Platform         0
Year             0
Genre            0
Publisher       36
NA_Sales         0
EU_Sales         0
JP_Sales         0
Other_Sales      0
Global_Sales     0
dtype: int64

In [48]:
print(vgs_clean[vgs_clean['Publisher'].isnull()].Name)

1662          Shrek / Shrek 2 2-in-1 Gameboy Advance Video
2222                                    Bentley's Hackpack
3159     Nicktoons Collection: Game Boy Advance Video V...
3166     SpongeBob SquarePants: Game Boy Advance Video ...
3766     SpongeBob SquarePants: Game Boy Advance Video ...
4526     The Fairly Odd Parents: Game Boy Advance Video...
4635     The Fairly Odd Parents: Game Boy Advance Video...
5647     Cartoon Network Collection: Game Boy Advance V...
6437              Sonic X: Game Boy Advance Video Volume 1
6562     Dora the Explorer: Game Boy Advance Video Volu...
6648     Cartoon Network Collection: Game Boy Advance V...
6849        All Grown Up!: Game Boy Advance Video Volume 1
7208     Nicktoons Collection: Game Boy Advance Video V...
7470     SpongeBob SquarePants: Game Boy Advance Video ...
7953                      Thomas the Tank Engine & Friends
8330       Dragon Ball GT: Game Boy Advance Video Volume 1
8341     Codename: Kids Next Door: Game Boy Advance Vid.

## Due by tomorrow morning:
### If we can find a code that makes our life easier to insert the found publisher online
search by Name: Enter cell# >> insert {Publisher}

In [30]:
vgs.head(10)

Rank                       Name Platform    Year         Genre Publisher  \
0     1                 Wii Sports      Wii  2006.0        Sports  Nintendo   
1     2          Super Mario Bros.      NES  1985.0      Platform  Nintendo   
2     3             Mario Kart Wii      Wii  2008.0        Racing  Nintendo   
3     4          Wii Sports Resort      Wii  2009.0        Sports  Nintendo   
4     5   Pokemon Red/Pokemon Blue       GB  1996.0  Role-Playing  Nintendo   
5     6                     Tetris       GB  1989.0        Puzzle  Nintendo   
6     7      New Super Mario Bros.       DS  2006.0      Platform  Nintendo   
7     8                   Wii Play      Wii  2006.0          Misc  Nintendo   
8     9  New Super Mario Bros. Wii      Wii  2009.0      Platform  Nintendo   
9    10                  Duck Hunt      NES  1984.0       Shooter  Nintendo   

   NA_Sales  EU_Sales  JP_Sales  Other_Sales  Global_Sales  
0     41.49     29.02      3.77         8.46         82.74  
1     29.08      3.58      6.81         0.77         40.24  
2     15.85     12.88      3.79         3.31         35.82  
3     15.75     11.01      3.28         2.96         33.00  
4     11.27      8.89     10.22         1.00         31.37  
5     23.20      2.26      4.22         0.58         30.26  
6     11.38      9.23      6.50         2.90         30.01  
7     14.03      9.20      2.93         2.85         29.02  
8     14.59      7.06      4.70         2.26         28.62  
9     26.93      0.63      0.28         0.47         28.31

In [24]:
vgs.describe()

Rank          Year      NA_Sales      EU_Sales      JP_Sales  \
count  16598.000000  16327.000000  16598.000000  16598.000000  16598.000000   
mean    8300.605254   2006.406443      0.264667      0.146652      0.077782   
std     4791.853933      5.828981      0.816683      0.505351      0.309291   
min        1.000000   1980.000000      0.000000      0.000000      0.000000   
25%     4151.250000   2003.000000      0.000000      0.000000      0.000000   
50%     8300.500000   2007.000000      0.080000      0.020000      0.000000   
75%    12449.750000   2010.000000      0.240000      0.110000      0.040000   
max    16600.000000   2020.000000     41.490000     29.020000     10.220000   

        Other_Sales  Global_Sales  
count  16598.000000  16598.000000  
mean       0.048063      0.537441  
std        0.188588      1.555028  
min        0.000000      0.010000  
25%        0.000000      0.060000  
50%        0.010000      0.170000  
75%        0.040000      0.470000  
max       10.570000     82.740000

In [9]:
vgs['Publisher'].nunique()

578

In [10]:
vgs['Platform'].unique()

array(['Wii', 'NES', 'GB', 'DS', 'X360', 'PS3', 'PS2', 'SNES', 'GBA',
       '3DS', 'PS4', 'N64', 'PS', 'XB', 'PC', '2600', 'PSP', 'XOne', 'GC',
       'WiiU', 'GEN', 'DC', 'PSV', 'SAT', 'SCD', 'WS', 'NG', 'TG16',
       '3DO', 'GG', 'PCFX'], dtype=object)

In [13]:
vgs['Genre'].nunique()

12

In [14]:
vgs['Genre'].unique()

array(['Sports', 'Platform', 'Racing', 'Role-Playing', 'Puzzle', 'Misc',
       'Shooter', 'Simulation', 'Action', 'Fighting', 'Adventure',
       'Strategy'], dtype=object)

##### Data type: Sales of certain video games.
##### 
- Rank - Ranking of overall sales.

- Name - The games name

- Platform - Platform of the games release (i.e. PC,PS4, etc.)

- Year - Year of the game's release

- Genre - Genre of the game

- Publisher - Publisher of the game

- NA_Sales - Sales in North America (in millions)

- EU_Sales - Sales in Europe (in millions)

- JP_Sales - Sales in Japan (in millions)

- Other_Sales - Sales in the rest of the world (in millions)

- Global_Sales - Total worldwide sales.


- (What is the time period that this data is conducting?)


### 1- Read the data
- Done
### 2- Explore the data
- In-progress
### 3- Clean the data 
- Work on cleaning Year (271) & Publisher (58)
- 
- Make the Rank our main index

### 4- Visualize the data
- 
### 5- Questions to ask (total of 9)
#### Hint question:  (A) What region do you think has the most sales? 

- (S) What genre is sold the most in each region? over the years? (Genre, NA, EU, JP, Other, Global_Sales)
- (S) What game has the highest ranking for the last 5 years in each region? (Year, Rank, NA_Sales, ..., etc)
- (S) Which publisher released highest ranking games from 2010 to 2020? (Publisher, Rank, Year)

- (F) Which Platform has the highest games' sales?(Platform, Frequency)
- (F) What is the average sales of each genre worldwide? (Genre, Global_Sales)
- (F) What is the average sales of action games worldwide in 2020? (Genre, Global_Sales)
- (F) What is the most common genre that has been released? (Genre, Frequency)

- (A) What year has the most released games? (Year, Frequency)
- (A) What game has the highest ranking worldwide, games released since 2010? (Year, Rank)
- (A) Which publisher has prodcued the highest number of games? (Publisher, Game)


Next step: 
- Drop the rest of the missing publishers IF we cannot find a code to put in the publishers
- More explore + comments + titles
- Visualization?? 
- Go over questions, delete those who can't be answered + hard to answer
- Draw a conclusion
- Get in the Kaggle webpage of the code and gather more ideas
- Make sure we all understand every part of the code + presentation
- Change questions that have specific year 